reactive/jenkins.py

```diff
@@ -230,3 +248,34 @@ def update_nrpe_config(nagios):
 def stop():
     service_stop("jenkins")
     status_set("maintenance", "Jenkins stopped")
+
+
+@when("reset.jenkins.storage")
+def reset_jenkins_storage():
+    status_set("maintenance", "Configuring Jenkins Storage")
+
+    jenkins_installed = get_state("apt.installed.jenkins")
+    if jenkins_installed:
+        service_stop('jenkins')
```

Comment:
Indeed I think this issue is related to how we (don't) handle data migration in case storage is attached post-deployment. See also my reply to the other comment below.

If we had support for transparent data migration I think your concerns would be mitigated. It's true that we'd have to "unnecessarily" copy over a few files (that should be real fast) and restart the service, but this sounds a negligible cost from the user perspective (probably a few seconds altogether on regular hardware).

The benefit would be that we would consolidate code to have a single flow and simplify testing and maintenance. In these cases, I'd tend to prefer to pay the cost of a little overhead and get simpler code in return.